compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6713,9 +6711,12 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
                     BLangExpression restArg = iExpr.restArgs.get(j);
                     BType memType = j < tupleMemCount ? tupleMemberTypes.get(j) : tupleRestType;
                     checkTypeParamExpr(restArg, this.env, memType, true);
-                    if (restType != symTable.semanticError && this.resultType == symTable.semanticError) {
-                        restType = this.resultType;
-                    }
+                    restType = getRestType(restType);
+                }
+            } else {
+                for (BLangExpression restArg : iExpr.restArgs) {
+                    checkExpr(restArg, this.env);
```

Comment:
Let's pass `symTable.semanticError` as the expected type.

Reply:
Changed